<!DOCTYPE html>
<html lang="en"><head>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/tabby.min.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-c3c2c9e745155556954bc4da23476b10.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.32">

  <meta name="author" content="Philipp Pelz">
  <title>ECLIPSE Presentations – Data Science for Electron Microscopy  Lecture 7: Gaussian Processes 2</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/theme/quarto-ce804447b3af982f2c55816970ecc9a3.css">
  <link rel="stylesheet" href="custom.css">
  <link href="../../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-dark">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
    <h1 class="title"><p>Data Science for Electron Microscopy<br> Lecture 7: Gaussian Processes 2</p></h1>
    
  <div class="quarto-title-authors">
    <div class="quarto-title-author">
  <div class="quarto-title-author-name">
  Philipp Pelz 
  </div>
                <p class="quarto-title-affiliation">
                FAU Erlangen-Nürnberg
              </p>
          </div>
    </div>
  
    <div class="footer-logos1">
    <img src="logos/FAU.png" alt="FAU Logo" width="20%">
    <img src="logos/imn.png" alt="IMN Logo" width="20%">
    <img src="logos/cenem.png" alt="CENEM Logo" width="20%">
    <img src="logos/erc.jpg" alt="Elettra Logo" width="20%">
  </div>
  </section>
<section>
<section id="bayesian-optimization-active-learning" class="title-slide slide level1 center">
<h1>Bayesian Optimization &amp; Active Learning</h1>

</section>
<section id="introduction" class="slide level2">
<h2>Introduction</h2>
<ul>
<li>Bayesian Optimization is a method for optimizing black-box functions that are expensive to evaluate.</li>
<li>Useful for tuning hyperparameters in machine learning models.</li>
</ul>
</section>
<section id="gold-mining-analogy" class="slide level2">
<h2>Gold Mining Analogy</h2>
<ul>
<li>Goal: Find the maximum gold content along a line with minimal drillings.</li>
<li>Two objectives:
<ol type="1">
<li><strong>Active Learning</strong>: Estimate gold distribution accurately.</li>
<li><strong>Bayesian Optimization</strong>: Find the location of maximum gold content.</li>
</ol></li>
</ul>

<img data-src="./img/GT.svg" class="r-stretch quarto-figure-center"><p class="caption">Initially, we have no idea about the gold distribution. We can learn the gold distribution by drilling at different locations. However, this drilling is costly. Thus, we want to minimize the number of drillings required while still finding the location of maximum gold quickly.</p></section>
<section id="active-learning" class="slide level2">
<h2>Active Learning</h2>
<ul>
<li>Aim: Minimize labeling costs while maximizing modeling accuracy.</li>
<li>Strategy: Label the point with the highest model uncertainty (variance).</li>
<li>Use Gaussian Process (GP) as a surrogate model for uncertainty estimates. <img data-src="./img/prior2posterior.png"></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Each new data point updates our surrogate model, moving it closer to the ground truth. The black line and the grey shaded region indicate the mean μ and uncertainty μ±σ in our gold distribution estimate before and after drilling.</li>
</ul>
</section>
<section class="slide level2">


<img data-src="./img/0.png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/1.png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/2.png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/3.png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/4.png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/5.png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/6.png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/7.png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/8.png" class="r-stretch"></section>
<section id="bayesian-optimization" class="slide level2">
<h2>Bayesian Optimization</h2>
<ul>
<li>Aim: Find the maximum of an unknown function efficiently.</li>
<li>Balance exploration (unknown regions) and exploitation (known high-value regions).</li>
<li>Key component: Acquisition function.</li>
</ul>
<h3 id="to-solve-this-problem-we-will-follow-the-following-algorithm">To solve this problem, we will follow the following algorithm:</h3>
<ol type="1">
<li>We first choose a surrogate model for modeling the true function f and define its prior.</li>
<li>Given the set of observations (function evaluations), use Bayes rule to obtain the posterior.</li>
<li>Use an acquisition function α(x), which is a function of the posterior, to decide the next sample point</li>
<li>Add newly sampled data to the set of observations and goto step #2 till convergence or budget elapses.</li>
</ol>
</section>
<section id="surrogate-models-and-gaussian-processes" class="slide level2">
<h2>Surrogate Models and Gaussian Processes</h2>
<ul>
<li>Surrogate models estimate the unknown function.</li>
<li>GPs are flexible and provide uncertainty estimates.</li>
<li>Update the surrogate model using Bayes’ rule after each evaluation.</li>
</ul>
</section>
<section id="acquisition-functions" class="slide level2">
<h2>Acquisition Functions</h2>
<h3 id="probability-of-improvement-pi">Probability of Improvement (PI)</h3>
<ul>
<li>Chooses the point with the highest probability of improvement over the current best.</li>
<li>Mathematically, we write the selection of next point as follows: <span class="math inline">\(x_{t+1} = argmax(\alpha_{PI}(x)) = argmax(P(f(x) \geq (f(x^+) +\epsilon)))\)</span></li>
<li><span class="math inline">\(\begin{aligned}
      x_{t+1} &amp; = argmax(\alpha_{PI}(x))\\
      &amp; = argmax(P(f(x) \geq (f(x^+) +\epsilon)))
      \end{aligned}\)</span></li>
<li>we are just finding the upper-tail probability (or the CDF) of the surrogate posterior. Moreover, if we are using a GP as a surrogate the expression above converts to</li>
<li><span class="math inline">\(x^+ = \text{argmax}_{x_i \in x_{1:t}}f(x_i)\)</span></li>
</ul>
</section>
<section class="slide level2">


<img data-src="./img/density_pi.png" class="r-stretch"></section>
<section id="intuition-behind-ϵ-in-pi-ϵ-0.075" class="slide level2">
<h2>Intuition behind ϵ in PI: ϵ = 0.075</h2>

<img data-src="./img/0 (1).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/1 (1).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/2 (1).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/3 (1).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/4 (1).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/5 (1).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/6 (1).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/7 (1).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/8 (1).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/9 (1).png" class="r-stretch"></section>
<section class="slide level2">

<ul>
<li>Looking at the graph above, we see that we reach the global maxima in a few iterations .</li>
<li>Our surrogate possesses a large uncertainty in x∈[2,4] in the first few iterations</li>
<li>The acquisition function initially exploits regions with a high promise , which leads to high uncertainty in the region x∈[2,4].</li>
<li>This observation also shows that we do not need to construct an accurate estimate of the black-box function to find its maximum.</li>
</ul>
</section>
<section id="intuition-behind-ϵ-in-pi-ϵ-0.3" class="slide level2">
<h2>Intuition behind ϵ in PI: ϵ = 0.3</h2>

<img data-src="./img/0 (2).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/1 (2).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/2 (2).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/3 (2).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/4 (2).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/5 (2).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/6 (2).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/7 (2).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/8 (2).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/9 (2).png" class="r-stretch"></section>
<section class="slide level2">

<ul>
<li>We see that we made things worse!</li>
<li>Our model now uses ϵ=3, and we are unable to exploit when we land near the global maximum. Moreover, with high exploration, the setting becomes similar to active learning.</li>
<li>Our quick experiments above help us conclude that ϵ controls the degree of exploration in the PI acquisition function.</li>
</ul>
</section>
<section id="expected-improvement-introduction" class="slide level2">
<h2>Expected Improvement: Introduction</h2>
<ul>
<li>Probability of improvement considers <em>how likely</em> an improvement is.</li>
<li>Expected Improvement (EI) considers <em>how much</em> we can improve.</li>
<li><strong>Key Idea</strong>: Choose the next query point with the highest expected improvement over the current max <span class="math inline">\(f(x^+)\)</span>.</li>
</ul>
</section>
<section id="ei-mathematical-formulation" class="slide level2">
<h2>EI: Mathematical Formulation</h2>
<ul>
<li><strong>Equation</strong>: <span class="math inline">\(x_{t+1} = \arg\min_x \mathbb{E} \left( ||h_{t+1}(x) - f(x^\star) || \ | \ \mathcal{D}_t \right)\)</span></li>
<li><strong>Components</strong>:
<ul>
<li><span class="math inline">\(f\)</span>: Actual ground truth function.</li>
<li><span class="math inline">\(h_{t+1}\)</span>: Posterior mean of the surrogate at <span class="math inline">\(t+1^{th}\)</span> timestep.</li>
<li><span class="math inline">\(\mathcal{D}_t\)</span>: Training data.</li>
<li><span class="math inline">\(x^\star\)</span>: Actual position where <span class="math inline">\(f\)</span> takes the maximum value.</li>
</ul></li>
</ul>
</section>
<section id="ei-mockus-acquisition-function" class="slide level2">
<h2>EI: Mockus’ Acquisition Function</h2>
<ul>
<li><strong>Equation</strong>: <span class="math inline">\(x_{t+1} = \mathrm{argmax}_x \mathbb{E} \left( {max} \{ 0, \ h_{t+1}(x) - f(x^+) \} \ | \ \mathcal{D}_t \right)\)</span></li>
<li><strong>Components</strong>:
<ul>
<li>$ f(x^+) $: Maximum value encountered so far.</li>
</ul></li>
</ul>
<h3 id="ei-analytical-expression-for-gp-surrogate">EI: Analytical Expression for GP Surrogate</h3>
<ul>
<li><strong>Equation</strong>: <span class="math inline">\(EI(x)=\begin{cases}(\mu_t(x) - f(x^+) - \epsilon)\Phi(Z) + \sigma_t(x)\phi(Z), &amp; \text{if}\ \sigma_t(x) &gt; 0\\ 0, &amp; \text{if}\ \sigma_t(x) = 0
\end{cases}\)</span></li>
</ul>
<p><span class="math inline">\(Z= \frac{\mu_t(x) - f(x^+) - \epsilon}{\sigma_t(x)}\)</span></p>
</section>
<section id="ei-when-is-ei-high" class="slide level2">
<h2>EI: When is EI High?</h2>
<ul>
<li>EI is high when:
<ul>
<li>The expected value of <span class="math inline">\(\mu_t(x) - f(x^+)\)</span> is high.</li>
<li>The uncertainty <span class="math inline">\(\sigma_t(x)\)</span> around a point is high.</li>
</ul></li>
</ul>
</section>
<section id="ei-moderating-exploration-with-epsilon" class="slide level2">
<h2>EI: Moderating Exploration with <span class="math inline">\(\epsilon\)</span></h2>
<ul>
<li>Adjusting <span class="math inline">\(\epsilon\)</span> moderates exploration.</li>
<li><strong>Examples</strong>:
<ul>
<li><span class="math inline">\(\epsilon = 0.01\)</span>: Close to the global maxima in few iterations.</li>
<li><span class="math inline">\(\epsilon = 0.3\)</span>: More exploration, less exploitation near the global maxima.</li>
<li><span class="math inline">\(\epsilon = 3\)</span>: Too much exploration, quick reach near the global maxima, less exploitation.</li>
</ul></li>
</ul>
</section>
<section id="ei-visualizations" class="slide level2">
<h2>EI: Visualizations</h2>
</section>
<section id="intuition-behind-ϵ-in-ei-ϵ-0.3" class="slide level2">
<h2>Intuition behind ϵ in EI: ϵ = 0.3</h2>

<img data-src="./img/0 (4).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/1 (4).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/2 (4).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/3 (4).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/4 (4).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/5 (4).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/6 (4).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/7 (4).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/8 (4).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/9 (4).png" class="r-stretch"></section>
<section id="intuition-behind-ϵ-in-pi-ϵ-0.3-1" class="slide level2">
<h2>Intuition behind ϵ in PI: ϵ = 0.3</h2>

<img data-src="./img/0 (5).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/1 (5).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/2 (5).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/3 (5).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/4 (5).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/5 (5).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/6 (5).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/7 (5).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/8 (5).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/9 (5).png" class="r-stretch"></section>
<section id="intuition-behind-ϵ-in-pi-ϵ-3" class="slide level2">
<h2>Intuition behind ϵ in PI: ϵ = 3</h2>

<img data-src="./img/0 (6).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/1 (6).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/2 (6).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/3 (6).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/4 (6).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/5 (6).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/6 (6).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/7 (6).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/8 (6).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/9 (6).png" class="r-stretch"></section>
<section id="thompson-sampling" class="slide level2">
<h2>Thompson Sampling</h2>
<ul>
<li>Samples a function from the posterior and optimizes it.</li>
<li>Balances exploration and exploitation naturally.</li>
</ul>

<img data-src="./img/thompson.svg" class="r-stretch"></section>
<section id="intuition-behind-thompson-sampling" class="slide level2">
<h2>Intuition behind Thompson Sampling</h2>

<img data-src="./img/0 (7).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/1 (7).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/2 (7).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/3 (7).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/4 (7).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/5 (7).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/6 (7).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/7 (7).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/8 (7).png" class="r-stretch"></section>
<section class="slide level2">


<img data-src="./img/9 (7).png" class="r-stretch"></section>
<section id="hyperparameter-tuning" class="slide level2">
<h2>Hyperparameter Tuning</h2>
<ul>
<li>Common use of Bayesian Optimization.</li>
<li>Examples: SVM, Random Forest, Neural Networks.</li>
<li>Bayesian Optimization efficiently searches hyperparameter space.</li>
</ul>
</section>
<section id="example-svm-hyperparameter-tuning" class="slide level2">
<h2>Example: SVM Hyperparameter Tuning</h2>
<ul>
<li>Optimize SVM hyperparameters ( ) and ( C ) on a dataset.</li>
<li>Compare acquisition functions (PI, EI, UCB).</li>
</ul>
</section>
<section id="summary" class="slide level2">
<h2>Summary</h2>
<ul>
<li>Bayesian Optimization is powerful for optimizing expensive black-box functions.</li>
<li>Key elements: Surrogate model (GP), Acquisition functions (PI, EI, UCB).</li>
<li>Applications in hyperparameter tuning for ML models.</li>
</ul>

<img data-src="./img/comp3d.svg" class="r-stretch"></section>
<section class="slide level2">

</section></section>
<section>
<section id="deep-kernel-learning---combining-neural-networks-and-gaussian-processes" class="title-slide slide level1 center">
<h1>Deep Kernel Learning - Combining Neural Networks and Gaussian Processes</h1>

</section>
<section id="the-big-question" class="slide level2">
<h2>The Big Question</h2>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>MacKay’s Question (1998)</strong></p>
</div>
<div class="callout-content">
<blockquote>
<p>“How can Gaussian processes possibly replace neural networks? Have we thrown the baby out with the bathwater?”</p>
</blockquote>
<ul>
<li>Neural networks: Many design choices, lack of principled framework</li>
<li>Gaussian processes: Flexible, interpretable, principled learning</li>
<li><strong>Can we combine the best of both worlds?</strong></li>
</ul>
</div>
</div>
</div>
</section>
<section id="the-evolution-of-ml-paradigms" class="slide level2">
<h2>The Evolution of ML Paradigms</h2>
<div class="columns">
<div class="column" style="width:50%;">
<h3 id="neural-networks">Neural Networks</h3>
<ul>
<li><strong>Finite</strong> adaptive basis functions</li>
<li>Multiple layers of highly adaptive features</li>
<li>Automatic representation discovery</li>
<li>Inductive biases for specific domains</li>
</ul>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Key Insight</strong></p>
</div>
<div class="callout-content">
<p>Neural networks can automatically discover meaningful representations in high-dimensional data</p>
</div>
</div>
</div>
</div><div class="column" style="width:50%;">
<h3 id="gaussian-processes">Gaussian Processes</h3>
<ul>
<li><strong>Infinite</strong> fixed basis functions</li>
<li>Non-parametric flexibility</li>
<li>Automatic complexity calibration</li>
<li>Uncertainty quantification</li>
</ul>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Key Insight</strong></p>
</div>
<div class="callout-content">
<p>GPs with expressive kernels can discover rich structure without human intervention</p>
</div>
</div>
</div>
</div></div>
</section>
<section id="the-deep-kernel-learning-idea" class="slide level2">
<h2>The Deep Kernel Learning Idea</h2>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Core Concept</strong></p>
</div>
<div class="callout-content">
<p>Transform the inputs of a base kernel with a deep architecture to create <strong>scalable expressive closed-form kernels</strong></p>
</div>
</div>
</div>
<div class="columns">
<div class="column" style="width:60%;">
<h3 id="mathematical-formulation">Mathematical Formulation</h3>
<p><span class="math display">\[k(\mathbf{x}_i, \mathbf{x}_j | \boldsymbol{\theta}) \rightarrow k(g(\mathbf{x}_i, \mathbf{w}), g(\mathbf{x}_j, \mathbf{w}) | \boldsymbol{\theta}, \mathbf{w})\]</span></p>
<p>Where:</p>
<ul>
<li><span class="math inline">\(g(\mathbf{x}, \mathbf{w})\)</span> = deep architecture (CNN, DNN)</li>
<li><span class="math inline">\(k(\cdot, \cdot | \boldsymbol{\theta})\)</span> = base kernel (RBF, Spectral Mixture)</li>
<li><span class="math inline">\(\boldsymbol{\gamma} = \{\mathbf{w}, \boldsymbol{\theta}\}\)</span> = all parameters</li>
</ul>
</div><div class="column" style="width:40%;">
<h3 id="key-benefits">Key Benefits</h3>
<ul>
<li><strong>Scalable</strong>: <span class="math inline">\(\mathcal{O}(n)\)</span> training, <span class="math inline">\(\mathcal{O}(1)\)</span> prediction</li>
<li><strong>Expressive</strong>: Combines deep features with kernel flexibility</li>
<li><strong>Non-parametric</strong>: Automatic complexity calibration</li>
<li><strong>Unified</strong>: Joint learning through GP marginal likelihood</li>
</ul>
</div></div>

<img data-src="./img/kernelnetwork.png" class="r-stretch quarto-figure-center"><p class="caption">Deep Kernel Learning</p></section>
<section id="deep-kernel-architecture" class="slide level2">
<h2>Deep Kernel Architecture</h2>
<div class="columns">
<div class="column" style="width:50%;">
<h3 id="network-structure">Network Structure</h3>
<ol type="1">
<li><strong>Input Layer</strong>: Raw data <span class="math inline">\(\mathbf{x}\)</span></li>
<li><strong>Hidden Layers</strong>: Deep transformation <span class="math inline">\(g(\mathbf{x}, \mathbf{w})\)</span></li>
<li><strong>Kernel Layer</strong>: Infinite basis functions via GP</li>
<li><strong>Output</strong>: Probabilistic predictions</li>
</ol>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Infinite Hidden Units</strong></p>
</div>
<div class="callout-content">
<p>The GP with base kernel provides an infinite number of basis functions in the final layer</p>
</div>
</div>
</div>
</div><div class="column" style="width:50%;">
<h3 id="learning-objective">Learning Objective</h3>
<p>Maximize the <strong>marginal likelihood</strong>:</p>
<p><span class="math display">\[\log p(\mathbf{y} | \boldsymbol{\gamma}, X) \propto -[\mathbf{y}^{\top}(K_{\boldsymbol{\gamma}}+\sigma^2 I)^{-1}\mathbf{y} + \log|K_{\boldsymbol{\gamma}} + \sigma^2 I|]\]</span></p>
<p><strong>Gradients via chain rule</strong>: <span class="math display">\[\frac{\partial \mathcal{L}}{\partial \mathbf{w}} = \frac{\partial \mathcal{L}}{\partial K_{\boldsymbol{\gamma}}} \frac{\partial K_{\boldsymbol{\gamma}}}{\partial g(\mathbf{x},\mathbf{w})} \frac{\partial g(\mathbf{x},\mathbf{w})}{\partial \mathbf{w}}\]</span></p>
</div></div>
</section>
<section id="base-kernels-rbf-vs-spectral-mixture" class="slide level2">
<h2>Base Kernels: RBF vs Spectral Mixture</h2>
<div class="columns">
<div class="column" style="width:50%;">
<h3 id="rbf-kernel">RBF Kernel</h3>
<p><span class="math display">\[k_{\text{RBF}}(\mathbf{x}, \mathbf{x}') = \exp\left(-\frac{1}{2} \frac{\|\mathbf{x}-\mathbf{x}'\|^2}{\ell^2}\right)\]</span></p>
<p><strong>Properties</strong>:</p>
<ul>
<li>Single length-scale parameter <span class="math inline">\(\ell\)</span></li>
<li>Smooth, stationary</li>
<li>Limited expressiveness</li>
</ul>
</div><div class="column" style="width:50%;">
<h3 id="spectral-mixture-kernel">Spectral Mixture Kernel</h3>
<p><span class="math display">\[k_{\text{SM}}(\mathbf{x}, \mathbf{x}' | \boldsymbol{\theta}) = \sum_{q=1}^Q a_q \frac{|\Sigma_q|^{1/2}}{(2\pi)^{D/2}} \exp\left(-\frac{1}{2} \|\Sigma_q^{1/2} (\mathbf{x}-\mathbf{x}')\|^2\right) \cos\langle \mathbf{x}-\mathbf{x}', 2\pi \boldsymbol{\mu}_q \rangle\]</span></p>
<p><strong>Properties</strong>:</p>
<ul>
<li>Multiple components <span class="math inline">\(Q\)</span></li>
<li>Quasi-periodic structure</li>
<li>Much more expressive</li>
</ul>
</div></div>
</section>
<section id="scalability-kiss-gp" class="slide level2">
<h2>Scalability: KISS-GP</h2>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>The Scalability Challenge</strong></p>
</div>
<div class="callout-content">
<p>Standard GPs: <span class="math inline">\(\mathcal{O}(n^3)\)</span> complexity <strong>Goal</strong>: Linear scaling <span class="math inline">\(\mathcal{O}(n)\)</span></p>
</div>
</div>
</div>
<h3 id="kiss-gp-approximation">KISS-GP Approximation</h3>
<p><span class="math display">\[K_{\boldsymbol{\gamma}} \approx M K^{\text{deep}}_{U,U} M^{\top} := K_{\text{KISS}}\]</span></p>
<p>Where:</p>
<ul>
<li><span class="math inline">\(M\)</span> = sparse interpolation matrix (4 non-zero entries per row)</li>
<li><span class="math inline">\(K_{U,U}\)</span> = covariance over inducing points <span class="math inline">\(U\)</span></li>
<li>Kronecker + Toeplitz structure for fast MVMs</li>
</ul>
<h3 id="computational-benefits">Computational Benefits</h3>
<ul>
<li><strong>Training</strong>: <span class="math inline">\(\mathcal{O}(n + h(m))\)</span> where <span class="math inline">\(h(m) \approx \mathcal{O}(m)\)</span></li>
<li><strong>Prediction</strong>: <span class="math inline">\(\mathcal{O}(1)\)</span> per test point</li>
<li><strong>Memory</strong>: <span class="math inline">\(\mathcal{O}(n)\)</span> instead of <span class="math inline">\(\mathcal{O}(n^2)\)</span></li>
</ul>
</section>
<section id="experimental-results-uci-datasets" class="slide level2">
<h2>Experimental Results: UCI Datasets</h2>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Key Finding</strong></p>
</div>
<div class="callout-content">
<p>Deep Kernel Learning consistently outperforms both: - Standalone deep neural networks - Gaussian processes with expressive kernels</p>
</div>
</div>
</div>
<div class="columns">
<div class="column" style="width:60%;">
<h3 id="performance-comparison">Performance Comparison</h3>
<ul>
<li><strong>16 UCI regression datasets</strong></li>
<li><strong>2M+ training examples</strong> (Electric dataset)</li>
<li><strong>DKL-SM</strong> achieves best performance on most datasets</li>
<li><strong>Minimal runtime overhead</strong> (~10% additional cost)</li>
</ul>
</div><div class="column" style="width:40%;">
<h3 id="architecture-details">Architecture Details</h3>
<ul>
<li><strong>Small datasets</strong> (<span class="math inline">\(n &lt; 6,000\)</span>): <span class="math inline">\([d\text{-}1000\text{-}500\text{-}50\text{-}2]\)</span></li>
<li><strong>Large datasets</strong> (<span class="math inline">\(n &gt; 6,000\)</span>): <span class="math inline">\([d\text{-}1000\text{-}1000\text{-}500\text{-}50\text{-}2]\)</span></li>
<li><strong>SM kernel</strong>: <span class="math inline">\(Q=4\)</span> (small), <span class="math inline">\(Q=6\)</span> (large)</li>
</ul>
</div></div>
</section>
<section id="learned-representations" class="slide level2">
<h2>Learned Representations</h2>
<div class="columns">
<div class="column" style="width:50%;">
<h3 id="spectral-density-analysis">Spectral Density Analysis</h3>
<ul>
<li><strong>SM kernel</strong>: Discovers two peaks in frequency domain</li>
<li><strong>RBF kernel</strong>: Single Gaussian, misses important correlations</li>
<li><strong>Result</strong>: SM captures quasi-periodic structure better</li>
</ul>
</div><div class="column" style="width:50%;">
<h3 id="covariance-matrix-analysis">Covariance Matrix Analysis</h3>
<ul>
<li><strong>DKL kernels</strong>: Strong correlation for similar orientations</li>
<li><strong>Standard RBF</strong>: Diffuse correlations</li>
<li><strong>Metric learning</strong>: Learns orientation-aware similarity</li>
</ul>
</div><div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Visualization</strong></p>
</div>
<div class="callout-content">
<p>The learned metric correlates faces with similar rotation angles, overcoming Euclidean distance limitations</p>
</div>
</div>
</div>
</div>
</section>
<section id="scalability-analysis" class="slide level2">
<h2>Scalability Analysis</h2>
<div class="columns">
<div class="column" style="width:50%;">
<h3 id="training-time-scaling">Training Time Scaling</h3>
<ul>
<li><strong>Linear scaling</strong> with data size <span class="math inline">\(n\)</span></li>
<li><strong>Slope ≈ 1</strong> in log-log plot</li>
<li><strong>KISS-GP</strong> enables large-scale training</li>
</ul>
</div><div class="column" style="width:50%;">
<h3 id="runtime-comparison">Runtime Comparison</h3>
<ul>
<li><strong>DNN</strong>: ~7-5000s (depending on dataset size)</li>
<li><strong>DKL</strong>: ~10-5000s (minimal overhead)</li>
<li><strong>Additional cost</strong>: ~10% of DNN runtime</li>
</ul>
</div><div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Key Benefit</strong></p>
</div>
<div class="callout-content">
<p>Scalability enables learning from large datasets where expressive representations matter most</p>
</div>
</div>
</div>
</div>
</section>
<section id="step-function-recovery" class="slide level2">
<h2>Step Function Recovery</h2>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Challenge</strong></p>
</div>
<div class="callout-content">
<p>Recover step function with sharp discontinuities</p>
</div>
</div>
</div>
<div class="columns">
<div class="column" style="width:60%;">
<h3 id="problem-characteristics">Problem Characteristics</h3>
<ul>
<li><strong>Multiple discontinuities</strong></li>
<li><strong>Sharp changes</strong> in covariance structure</li>
<li><strong>Difficult for smooth kernels</strong></li>
</ul>
<h3 id="results">Results</h3>
<ul>
<li><strong>GP-RBF</strong>: Smooth, misses discontinuities</li>
<li><strong>GP-SM</strong>: Better, but still smooth</li>
<li><strong>DKL-SM</strong>: Accurately captures discontinuities with reasonable uncertainty</li>
</ul>
</div><div class="column" style="width:40%;">
<h3 id="key-advantage">Key Advantage</h3>
<p>DKL provides <strong>posterior predictive distributions</strong> useful for:</p>
<ul>
<li>Reinforcement learning</li>
<li>Bayesian optimization</li>
<li>Uncertainty quantification</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./img/syn_step_func.png" height="400"></p>
<figcaption>Step Function Recovery</figcaption>
</figure>
</div>
</div></div>
</section>
<section id="key-contributions" class="slide level2">
<h2>Key Contributions</h2>
<h3 id="scalable-deep-kernels">1. Scalable Deep Kernels</h3>
<ul>
<li>Linear scaling <span class="math inline">\(\mathcal{O}(n)\)</span> training</li>
<li><span class="math inline">\(\mathcal{O}(1)\)</span> prediction time</li>
<li>Retains non-parametric representation</li>
</ul>
<h3 id="expressive-power">2. Expressive Power</h3>
<ul>
<li>Combines deep architectures with kernel flexibility</li>
<li>Spectral mixture base kernels</li>
<li>Automatic complexity calibration</li>
</ul>
<h3 id="unified-learning">3. Unified Learning</h3>
<ul>
<li>Joint optimization through GP marginal likelihood</li>
<li>No separate pre-training required</li>
<li>Drop-in replacement for standard kernels</li>
</ul>
</section>
<section id="summary-1" class="slide level2">
<h2>Summary</h2>
<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Deep Kernel Learning Successfully Combines:</strong></p>
</div>
<div class="callout-content">
<h3 id="neural-networks-1">Neural Networks</h3>
<ul>
<li>Automatic representation discovery</li>
<li>Inductive biases for specific domains</li>
<li>Scalable training procedures</li>
</ul>
<h3 id="gaussian-processes-1">Gaussian Processes</h3>
<ul>
<li>Non-parametric flexibility</li>
<li>Uncertainty quantification</li>
<li>Principled learning framework</li>
</ul>
<h3 id="result">Result</h3>
<p><strong>Scalable, expressive, and principled</strong> machine learning approach that consistently outperforms both paradigms alone</p>
</div>
</div>
</div>
</section>
<section id="discussion-points" class="slide level2">
<h2>Discussion Points</h2>
<ol type="1">
<li>How does DKL compare to other kernel learning approaches?</li>
<li>What are the computational trade-offs?</li>
<li>When would you choose DKL over standalone DNNs or GPs?</li>
<li>How does the choice of base kernel affect performance?</li>
</ol>
</section>
<section id="why-automate-4dstem" class="slide level2">
<h2>Why automate 4D‑STEM?</h2>
<ul>
<li>High‑dimensional data cube → costly in <strong>dose</strong> &amp; <strong>time</strong><br>
</li>
<li>Human ROI selection = biased &amp; inefficient<br>
</li>
<li><strong>Active learning</strong> can steer the probe to where information content is highest</li>
</ul>
</section>
<section id="from-bayesian-optimisation-to-autonomous-stem" class="slide level2">
<h2>From Bayesian Optimisation&nbsp;to Autonomous STEM</h2>
<p>We have already covered</p>
<ol type="1">
<li><strong>Bayesian optimisation (BO)</strong><br>
</li>
<li><strong>Deep‑Kernel Learning (DKL)</strong></li>
</ol>
<p>Today we see <strong>both</strong> in action inside the microscope.</p>
<p><span class="math display">\[\begin{equation}
\operatorname{EI}(\mathbf{x}) = \big(\mu(\mathbf{x}) - y^{+} - \xi\big)\,\Phi\!\left(\dfrac{\mu(\mathbf{x})-y^{+}-\xi}{\sigma(\mathbf{x})}\right) + \sigma(\mathbf{x})\,\phi\!\left(\dfrac{\mu(\mathbf{x})-y^{+}-\xi}{\sigma(\mathbf{x})}\right)
\end{equation}\]</span></p>

<aside><div>
<p><span class="citation" data-cites="KalininAutomated_2022">Roccapriore, Kevin M. et al. <a href="#/references" role="doc-biblioref" onclick="">(2022), doi:<span>10.1021/acsnano.1c11118</span></a></span></p>
</div></aside></section></section>
<section id="dstem-primer" class="title-slide slide level1 center">
<h1>4D‑STEM primer</h1>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="roccapriore2022_images/page_001_img_001.png" style="width:70.0%"></p>
<figcaption>4D‑STEM schematic</figcaption>
</figure>
</div>

<aside><div>
<p><span class="citation" data-cites="KalininAutomated_2022">Roccapriore, Kevin M. et al. <a href="#/references" role="doc-biblioref" onclick="">(2022), doi:<span>10.1021/acsnano.1c11118</span></a></span></p>
</div></aside></section>

<section>
<section id="dkl-workflow-for-4dstem" class="title-slide slide level1 center">
<h1>DKL workflow for 4D‑STEM</h1>

<img data-src="roccapriore2022_images/page_002_img_001.png" style="width:45.0%" class="r-stretch quarto-figure-center" id="fig-dkl"><p class="caption">
Figure&nbsp;1: DKL workflow for 4D-STEM: learning (a), prediction (b), and measurement (c). Features are HAADF-STEM image patches; targets are scalarized diffraction patterns from patch centers.
</p><div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Key idea</strong></p>
</div>
<div class="callout-content">
<p>CNN‑based embedding → GP kernel → BO acquisition</p>
</div>
</div>
</div>
</section>
<section id="ground-truth-data-examples" class="slide level2">
<h2>Ground Truth Data Examples</h2>
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="roccapriore2022_images/page_005_img_001.png" style="width:100.0%"></p>
<figcaption>Ground truth 4D-STEM data: single-layer and bilayer graphene with defects. HAADF-STEM images show selected ronchigrams and local patches. CoM-based scalar quantities (CoMx, CoMy, angle, magnitude) encode local electric fields.</figcaption>
</figure>
</div>
</div><div class="column" style="width:50%;">
<p>Two 4D datasets demonstrate the approach:</p>
<ul>
<li><strong>Single-layer and bilayer graphene</strong> with defects and dopant atoms</li>
<li><strong>Diffraction patterns</strong> recorded at all pixel positions</li>
<li><strong>Center of mass (CoM)</strong> calculation from central beam</li>
<li><strong>Relative CoM shifts</strong> → local electric field computation</li>
<li><strong>Derived quantities</strong>: charge density and electric potential</li>
</ul>
</div></div>
<aside><div>
<p><span class="citation" data-cites="KalininAutomated_2022">Roccapriore, Kevin M. et al. <a href="#/references" role="doc-biblioref" onclick="">(2022), doi:<span>10.1021/acsnano.1c11118</span></a></span></p>
</div></aside></section>
<section id="damage-mitigation-strategy" class="slide level2">
<h2>Damage Mitigation Strategy</h2>
<p><strong>Intelligent sampling</strong> reduces beam damage through:</p>
<ul>
<li><strong>Specimen protection</strong>: Between training steps, specimen is blocked from electron irradiation</li>
<li><strong>Minimal sampling</strong>: Only a small fraction of available points are ever visited</li>
<li><strong>HAADF-STEM pre-scan</strong>: Complete image space access for the model</li>
<li><strong>Controlled dose</strong>: Small but unavoidable initial dose compared to full 4D acquisition</li>
</ul>

<aside><div>
<p><span class="citation" data-cites="KalininAutomated_2022">Roccapriore, Kevin M. et al. <a href="#/references" role="doc-biblioref" onclick="">(2022), doi:<span>10.1021/acsnano.1c11118</span></a></span></p>
</div></aside></section>
<section id="reconstruction-from-1-of-data" class="slide level2">
<h2>Reconstruction from <em>1 %</em> of data</h2>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="roccapriore2022_images/page_006_img_001.png" style="width:58.0%"></p>
<figcaption>4D-STEM DKL experimental results on Nion UltraSTEM100. Shows CoM magnitude (a) and angle (b) scalarizers with HAADF images, acquisition functions, predictions, and uncertainties at key steps. Red points mark visited locations. Scale bars: 1 nm.</figcaption>
</figure>
</div>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Tip</strong></p>
</div>
<div class="callout-content">
<p>DKL recovers CoM‑magnitude map with nanometre detail from &lt;1 % of pixels → <strong>30‑fold dose reduction</strong></p>
</div>
</div>
</div>

<aside><div>
<p><span class="citation" data-cites="KalininAutomated_2022">Roccapriore, Kevin M. et al. <a href="#/references" role="doc-biblioref" onclick="">(2022), doi:<span>10.1021/acsnano.1c11118</span></a></span></p>
</div></aside></section>
<section id="nanobeam-strain-mapping" class="slide level2">
<h2>Nanobeam strain mapping</h2>
<div class="columns">
<div class="column" style="width:60%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="roccapriore2022_images/page_006_img_001.png" style="width:100.0%"></p>
<figcaption>Ground truth NBED strain mapping and DKL exploration pathway. Shows Bragg distance scalarizer results with acquisition function preferring boundary measurements.</figcaption>
</figure>
</div>
</div><div class="column" style="width:40%;">
<p><strong>Nanobeam electron diffraction (NBED)</strong> approach:</p>
<ul>
<li><strong>Smaller convergence angle</strong> → diffraction discs</li>
<li><strong>Bragg disc centers</strong> scalarizer function<br>
</li>
<li><strong>First-order diffraction</strong> only considered</li>
<li><strong>Boundary preference</strong> in acquisition function</li>
<li><strong>Vacuum measurements</strong> for uncertainty reduction</li>
</ul>
</div></div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p><strong>Key insight</strong>: DKL learns to measure near boundaries where strain is highest, even without prior knowledge of material structure</p>
</div>
</div>
</div>
</section>
<section id="live-autonomous-experiment" class="slide level2">
<h2>Live autonomous experiment</h2>

<img data-src="roccapriore2022_images/page_008_img_001.png" style="width:90.0%" class="r-stretch quarto-figure-center"><p class="caption">DKL active learning on MnPS3 with DPC CoM scalarizer. Shows exploration pathway, predictions, and uncertainty at key steps. Periodic interference from sulfur vacancy generation. Scale bars: 5 and 2 nm.</p><div class="columns">
<div class="column" style="width:50%;">
<p><strong>MnPS₃ beam-sensitive material:</strong></p>
<ul>
<li><strong>Layered van der Waals 2D material</strong></li>
<li><strong>Sulfur vacancy generation</strong> from electron beam</li>
<li><strong>Hexagonal interference patterns</strong> in HAADF</li>
<li><strong>Defocus imaging</strong> (−40 nm) for atomic contrast</li>
<li><strong>3% sampling</strong> → 30× dose reduction</li>
</ul>
</div><div class="column" style="width:50%;">
<p><strong>DKL autonomous exploration:</strong></p>
<ul>
<li><strong>CoM-magnitude</strong> vs.&nbsp;<strong>CoM-angle</strong> scalarizer</li>
<li><strong>Model retraining</strong> every measurement (~2 s GPU)</li>
<li><strong>Boundary preference</strong> in acquisition function</li>
<li><strong>Structure-property learning</strong> from minimal data</li>
<li><strong>Real-time adaptation</strong> to specimen changes</li>
</ul>
</div></div>
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Tip</strong></p>
</div>
<div class="callout-content">
<p><strong>Key insight</strong>: DKL discovers ordered vacancy superstructures while protecting beam-sensitive specimens</p>
</div>
</div>
</div>
</section>
<section id="conclusions" class="slide level2">
<h2>Conclusions</h2>
<div class="columns">
<div class="column" style="width:60%;">
<p><strong>Active learning for 4D-STEM imaging:</strong></p>
<ul>
<li><strong>Deep kernel learning</strong> enables discovery of internal field behaviors</li>
<li><strong>Demonstrated</strong> on twisted bilayer graphene and MnPS₃ patterns</li>
<li><strong>Physics-based discovery</strong> of unique phenomena in quantum materials</li>
<li><strong>Extensible approach</strong> for strongly correlated materials</li>
</ul>
</div><div class="column" style="width:40%;">
<p><strong>Future opportunities:</strong></p>
<ul>
<li><strong>Pre-trained VAE weights</strong> for faster DKL training</li>
<li><strong>Physical deconvolution</strong> knowledge for latent space structuring</li>
<li><strong>Conditioning strategies</strong> for more directed physical search</li>
<li><strong>Interventional strategies</strong> via bootstrapping approaches</li>
</ul>
</div></div>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p><strong>Key insight</strong>: Active learning transforms 4D-STEM into an autonomous discovery platform for quantum materials research</p>
</div>
</div>
</div>
</section>
<section id="takeaways" class="slide level2">
<h2>Take‑aways</h2>
<ul>
<li><strong>DL‑guided BO</strong> turns the microscope into an <em>autonomous scientist</em><br>
</li>
<li>Massive <strong>dose/time savings</strong> without losing resolution<br>
</li>
<li>Works on <em>physics‑derived</em> targets (E‑field, strain, etc.)<br>
</li>
<li>Ready for integration with <strong>Edge/Cloud inference</strong> &amp; <strong>reinforcement control</strong></li>
</ul>
</section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>
<div id="refs" class="references csl-bib-body" role="list">
<div id="ref-KalininAutomated_2022" class="csl-entry" role="listitem">
Automated experiment in 4D-STEM: Exploring emergent physics and structural behaviors, ACS Nano, Kevin M. Roccapriore, Ondrej Dyck, Mark P. Oxley, Maxim Ziatdinov, &amp; Sergei V. Kalinin https://doi.org/<a href="https://doi.org/10.1021/acsnano.1c11118">10.1021/acsnano.1c11118</a>.
</div>
</div>
<ul>
<li><a href="https://distill.pub/2020/bayesian-optimization/">Distill Article on Bayesian Optimization</a></li>
</ul>
<script>
document.getElementById("marimo-frame").onload = function() {
    try {
        let iframeDoc = document.getElementById("marimo-frame").contentWindow.document;
        let marimoBadge = iframeDoc.querySelector("div.fixed.bottom-0.right-0.z-50");
        if (marimoBadge) {
            marimoBadge.style.display = "none";
            console.log("Marimo badge hidden successfully.");
        } else {
            console.log("Badge not found.");
        }
    } catch (error) {
        console.warn("Unable to modify iframe content due to CORS restrictions.");
    }
};
</script>
</section></section></div>

<div class="quarto-auto-generated-content" style="display: none;">
<p><img src="eclipse_logo_small.png" class="slide-logo"></p>
<div class="footer footer-default">
<p>©Philipp Pelz - FAU Erlangen-Nürnberg - Data Science for Electron Microscopy</p>
</div>
</div>

    </div>
  

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"right","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true,"width":"wide"},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: false,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'fade',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1920,

        height: 1080,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
            const codeEl = trigger.previousElementSibling.cloneNode(true);
            for (const childEl of codeEl.children) {
              if (isCodeAnnotation(childEl)) {
                childEl.remove();
              }
            }
            return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp("https:\/\/ECLIPSE-Lab\.github\.io\/public_presentations\/");
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>